src/Google.Api.Gax/Api/Gax/ServiceEndpointSettings.cs

```diff
@@ -60,5 +61,19 @@ public string Host
             Host = Host,
             Port = Port,
         };
+
+        /// <summary>
+        /// Creates a channel for the host and port in this object, with defaults for unspecified values.
+        /// </summary>
+        /// <param name="defaultHost">The host to use if <see cref="Host"/> is null. Must not be null.</param>
+        /// <param name="defaultPort">The port to use if <see cref="Port"/> is null.</param>
+        /// <param name="credentials">The credentials to use with the channel. Must not be null.</param>
+        /// <returns></returns>
+        public Channel CreateChannel(string defaultHost, int defaultPort, ChannelCredentials credentials)
```

Comment:
Could do, sure. If it weren't for it being a public class that users would interact with, I think it would still make sense where it is - but yes, I'm happy to move it to avoid confusion.